File: src/main/java/yaasl/server/event/EventBusConfig.java
package yaasl.server.event;

import net.engio.mbassy.bus.MBassador;
import net.engio.mbassy.bus.config.BusConfiguration;
import net.engio.mbassy.bus.config.Feature;
import net.engio.mbassy.bus.config.IBusConfiguration;
import net.engio.mbassy.bus.error.PublicationError;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class EventBusConfig {

    private Logger LOG = LoggerFactory.getLogger(getClass());

    private MBassador mBassador;

    @Bean
    public MBassador getMbassador() {
        if (mBassador == null) {
            IBusConfiguration busConfiguration = new BusConfiguration()
                    .addFeature(Feature.SyncPubSub.Default())
                    .addFeature(Feature.AsynchronousHandlerInvocation.Default())
                    .addFeature(Feature.AsynchronousMessageDispatch.Default())
                    .addPublicationErrorHandler((PublicationError error) -> {
                        LOG.error("Error during publish/subscribe dispatching", error.getCause());
                    });
            mBassador = new MBassador(busConfiguration);
        }
        return mBassador;
    }

}

File: src/main/kotlin/yaasl/server/event/EventBusConfig.kt
package yaasl.server.event

import net.engio.mbassy.bus.MBassador
import net.engio.mbassy.bus.config.BusConfiguration
import net.engio.mbassy.bus.config.Feature
import net.engio.mbassy.bus.error.PublicationError
import org.slf4j.LoggerFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class EventBusConfig {

    private val LOG = LoggerFactory.getLogger(javaClass)

    @Bean
    fun getMbassador(): MBassador<Any> {
        val busConfiguration = BusConfiguration()
                .addFeature(Feature.SyncPubSub.Default())
                .addFeature(Feature.AsynchronousHandlerInvocation.Default())
                .addFeature(Feature.AsynchronousMessageDispatch.Default())
                .addPublicationErrorHandler { error: PublicationError -> LOG.error("Error during publish/subscribe dispatching", error.cause) }
        return MBassador(busConfiguration)
    }

}
